_clients/javascript/helpers.md

```diff
@@ -0,0 +1,193 @@
+---
+layout: default
+title: Helper methods
+parent: JavaScript client
+nav_order: 2
+---
+
+# Helper methods
+
+This section is currently incomplete. It does not yet cover all helper methods and their options.
+{: .warning }
+
+## Bulk helper
+
+Running bulk requests can be complex because of the API structure. The bulk helper aims to provide a nicer developer experience around the Bulk API.
+
+### Usage
+
+The following code creates a bulk helper instance:
+
+```javascript
+const { Client } = require('@opensearch-project/opensearch')
+const documents = require('./docs.json')
+
+const client = new Client({ ... })
+
+const result = await client.helpers.bulk({
+  datasource: documents,
+  onDocument (doc) {
+    return {
+      index: { _index: 'example-index' }
+    }
+  }
+})
+
+console.log(result)
+```
+
+Bulk helper operations return an object with the following fields:
+
+```json
+{
+  total: number,
+  failed: number,
+  retry: number,
+  successful: number,
+  time: number,
+  bytes: number,
+  aborted: boolean
+}
+```
+
+When creating a new bulk helper instance, you can use the following configuration options.
+
+| Option | Data type | Description 
+| :--- | :--- | :---
+| `concurrency` | Integer | The number of requests to be executed in parallel. Optional. Default is 5. 
+| `datasource` | An array, async generator or a readable stream of strings or objects | Represents the documents you need to create, delete, index or update. Required.
+| `flushBytes` | Integer | Maximum bulk body size to send in bytes. Optional. Default is 5,000,000. 
+| `flushInterval` | Integer | Time in milliseconds to wait before flushing the body after the last document has been read. Optional. Default is 30,000.
+| `onDocument` | Function | A function to be invoked with each document in the given `datasource`. It returns the operation to be executed for this document. Optionally, the document can be manipulated for `create` and `index` operations by returning a new document as part of the function's result. Required.
+| `onDrop` | Function | A function to be invoked for every document that can’t be indexed after reaching the maximum amount of retries. Optional. Default is `noop`.
+| `refreshOnCompletion` | Boolean | Whether a refresh should be run on all affected indices at the end of the bulk operation. Optional. Default is false.
+| `retries` | Integer | The number of times an operation is retried before `onDrop` is called for that document. Optional. Defaults to the client's  `maxRetries` value.
+| `wait` | Integer | Time in milliseconds to wait before retrying an operation. Optional. Default is 5,000.
+
```

Comment:
Can we bubble the required params to the top of this list, and make it easier to see that they are required? One way of doing this is having a column called `required / default` that either states that param is required or a default value, if any, when it's optional:

 Option | Data type | Required / Default | Description 
| :--- | :--- | :--- | :---
| `datasource` | An array, async generator or a readable stream of strings or objects | REQUIRED | Represents the documents you need to create, delete, index or update.
| `concurrency` | Integer | 5 | The number of requests to be executed in parallel.
| `flushBytes` | Integer | 5,000,000. | Maximum bulk body size to send in bytes.

Thank you and great work :-)

Reply:
Done